### ISSUE TITLE
[Question] Evaluating performance of different operator combinations in the data hyper-cleaning example

### ISSUE THREAD

**Issue by Aprilsy-yy** [callous-youth/BOAT#37]
### Required prerequisites

- [x] I have read the documentation <https://boat.readthedocs.io/en/main/>.
- [x] I have searched the [Issue Tracker](https://github.com/callous-youth/BOAT/issues) to make sure this question has not already been answered.

### Question

**Body:**
**Environment**
OS: Linux 
Python: 3.12.13
PyTorch: 2.11.0+cu130 (via remote server)
Torchvision: 0.26.0

**What I’m trying to do**
I am reproducing the data_hyper_cleaning.py example to understand the practical impact of different operator combinations on convergence and final accuracy. Specifically, I am comparing various Hyper-gradient estimation operators (NGD, DI, GDA) against Lower-level approximation operators (RAD, CG, PTT) in the context of noisy label cleansing on MNIST.

**Repro steps**
I ran the following commands with extended iterations (60 and 100) to better observe the long-term convergence behavior:
python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op DI,NGD --na_op RAD --iterations 60
python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op GDA,NGD --na_op RAD --iterations 100
**Observed Performance & Analysis**
Based on my reproduction runs at 60 iterations, I observed distinct behaviors across different combinations:
Top Performer: The DI,NGD + RAD combination achieved the best overall results, with a final accuracy of 90.050 and the lowest validation loss of 0.350. It demonstrated the fastest convergence among all tested settings.
Steady Improvement: GDA,NGD + RAD reached an accuracy of 80.630 (val_loss 1.338). While it improved more slowly than the DI version, it continued to benefit from extending the run to 100 iterations (reaching 82.370).
Competitive Baseline: NGD + CG (78.750 acc) and NGD + RAD (77.070 acc) showed similar performance levels, with CG providing a slight edge in stability over long runs.
Weakest Link: The NGD + RAD,PTT setting was significantly less effective in this task, stagnating at an accuracy of 54.280 and a much higher validation loss of 2.147.
**Questions / Suggestions**
（1）Metric Preference: For the data hyper-cleaning task, should acc and val_loss be considered the primary benchmarks for comparing operator effectiveness, or are there more task-specific metrics (e.g., noise detection rate) you would recommend?
（2）Benchmark Subset: Does the BOAT team have a recommended "standard subset" of operator combinations that users should prioritize when reporting reproducible results in research?
（3）Expected Alignment: Does the superior performance of DI,NGD + RAD in these tests align with your internal benchmarks for this task, or could it be specific to the noisy MNIST setup used in this example?
